RIOT-OS/RIOT
# sys/net/dhcpv6: Fixes for MUD URL option

### Contribution description

This PR fixes a couple of mistakes I made in PR #15508 which added the Manufacturer Usage Description option (MUD, [RFC 8520](https://tools.ietf.org/html/rfc8520#section-10)) to the DHCPv6 client.

So far, the option was only included in Solicit Messages. According to [RFC 8415](https://tools.ietf.org/html/rfc8415#section-18.2.1), however, this is only permitted if it is "specifically allowed in the definition of individual options". I therefore removed the call of `_compose_mud_url_opt` from the `_solicit_servers` function and added it to `_request_renew_rebind` instead, including the option now in Request, Renew, and Rebind Messages. I updated the tests in `tests/gnrc_dhcpv6_client_6lbr` accordingly.

I also noticed that the naming of the Kconfig option I added in #15508 was incorrect. Setting the option `DHCPV6_CLIENT_MUD_URL` should now work as intended.

Furthermore, I rearranged some of the code, hopefully making the module a bit more efficient and robust. For example, the assertion of the MUD URL now happens in the `dhcpv6_client_init` function causing the application to crash immediately if something is wrong with the URL (rather than when a Request Message is sent).

Lastly I have noticed that the instructions for using the `tests-as-root` was outdated both in the readme files of  `tests/gnrc_dhcpv6_client_6lbr` and `tests/gnrc_dhcpv6_client`. I edited the two files in two separate commits which could also be merged into one (especially if this information should be missing in more readmes).

### Testing procedure

The changes to the client can be tested manually by using the updated tests provided in `tests/gnrc_dhcpv6_client_6lbr`. 

The Kconfig option can be tested, for instance, by first enabling Kconfig in the Makefile of `tests/gnrc_dhcpv6_client_6lbr` and then changing the MUD URL by using `make menuconfig`. If the URL is not `https://example.org` anymore, performing the tests with `sudo make test-as-root` should fail.

### Issues/PRs references

See also #15508.
